apiserver/facades/agent/uniter/networkinfo.go

```diff
@@ -168,25 +184,36 @@ func (n *NetworkInfoBase) getEgressForRelation(
 	return egress, errors.Trace(err)
 }
 
-// getEgressFromIngress returns a subnet corresponding to the first address
+// getEgressFromIngress returns subnets corresponding to the first address
 // (if available) in the input ingress address list.
 // There can be situations (observed for CAAS) where the preferred ingress
-// address is a FQDN for a load-balancer, which is intended to point at a
-// service that is not yet up.
-// If we cannot resolve the FQDN, log a warning and return a nil result.
+// address is a FQDN.
+// For these cases we log a warning and eschew egress determination.
 func (n *NetworkInfoBase) getEgressFromIngress(ingress []string) ([]string, error) {
```

Comment:
If the method always returns 0 or 1 egress CIDRs shouldn't it be instead named something `getOne/FirstEgressFromIngress` and return back a string instead of a slice?

Reply:
Its result is always used in a down-stream slice return, so it is smoother to assign directly rather than checking the result and making a slice at multiple call sites.

I actually made and then reverted a change to include multiple CIDRs because I didn't want to change behaviour in this patch, but I think it is something we should discuss.